### PR TITLE
adding ex1

### PR DESCRIPTION
**Pull request by AliRje82**

### CODE REVIEW

File: basic_integral/basic_integral.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#include <iostream>
+#include <math.h>
```
Comment (saeedafra) [inline basic_integral/basic_integral.cpp:2]: Any particular reason for using the .h file? cmath is standard c++ library which I guess is more reliable.

File: basic_integral/basic_integral.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+using namespace std;
+
+int main(){
+    cout << "Iterations ?";
+    int n;
```
Comment (saeedafra) [inline basic_integral/basic_integral.cpp:8]: Isn't it nice to have all declarations at the beginning of the function? Any reason for not doing so?

File: basic_integral/basic_integral.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    cin >> n;
+    int N=n;
```
Comment (saeedafra) [inline basic_integral/basic_integral.cpp:10]: Why would you duplicate variable n?

File: basic_integral/basic_integral.cpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    double delta ,sum;
+
+    for(int i=0; i<n ;i++){
+        sum=0;
+        delta=2 * M_PI/N;
```
Comment (saeedafra) [inline basic_integral/basic_integral.cpp:15]: delta doesn't depend on i. Why would you put it inside the loop?

File: basic_integral/basic_integral.cpp [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        for(int j=0 ; j<N;j++){
```
Comment (saeedafra) [inline basic_integral/basic_integral.cpp:16]: bit of punctuation helps here ;)

File: basic_integral/basic_integral.cpp [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            sum+=sin(delta *j + delta/2)*delta;
```
Comment (saeedafra) [inline basic_integral/basic_integral.cpp:17]: again, punctuation ( I mean spacing, etc)

Comment (saeedafra) [inline basic_integral/basic_integral.cpp:17]: multiplication by delta could as well be done after the loop for better speed.

**Review by saeedafra — CHANGES_REQUESTED** [review]
Changes requested. Please see the comments. Thanks

File: basic_integral/basic_integral.cpp (hunk 3 above)
Comment (AliRje82) [reply basic_integral/basic_integral.cpp:10]: its not duplicate they are different . 
n is Iteration times but N is upper limit of sigma

File: basic_integral/basic_integral.cpp [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    cin >> n;
+    
+    N=n; //Upper limit of sigma
+
+
+    for(int i=0; i<n ;i++){ // Loop iterate n times 
+        sum=0;
+        delta=2 * M_PI/N;
+
+        for(int j=0 ; j<N;j++) sum+=sin(delta *j + delta/2)*delta;
```
Comment (saeedafra) [inline basic_integral/basic_integral.cpp:21]: not the best style, but fine for now. We'll get back to this later.

File: astericks/astericks.cpp [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    cout << "number of rows?" << endl;
+    cin >> row;
+    cout <<"number of columns?" << endl;
+    cin >> col;
+
+    for(int i =0; i<row ; i++){
+
+        for(int j=0 ; j<col ;j++){
+
+            if(i==0 || i==row-1) cout << "*";
```
Comment (saeedafra) [inline astericks/astericks.cpp:17]: same here, not a good style. but fine for now.

**Review by saeedafra — APPROVED** [review]
Thanks for the good work. There are still some style concerns but that's all fine for now. Let's move on :)

**Pull request merged**